### PR TITLE
Further ailment improvements

### PR DESCRIPTION
**Pull request by Quotae**
I was directed to Issue #70, which was pointing to some lines that I added in my previous pull request which added the ailment box in the calcs tab. For some reason it was giving an error when you switch Calculations Mode to In Combat, but doesn't happen in Effective DPS. I fixed that, then along the way noticed a few things I should've done in the first pull request:

- Made Shock and Chill Ailment Thresholds be affected by Shock/Chill duration (I was warned about this but didn't get a chance to see those comments before it got merged)
- Separated Chill and Freeze in the code (not sure why I did it like I did in the first place lol)
- Added support for other types of damage to Chill (ie Soul Taker and Elementalist's Confluxes)

### CODE REVIEW

File: Modules/CalcOffence-3_0.lua [hunk 1]
```diff
@@ -2049,11 +2046,45 @@ function calcs.offence(env, actor, activeSkill)
 				output.ShockDurationMod = 1 + skillModList:Sum("INC", cfg, "EnemyShockDuration") / 100 + enemyDB:Sum("INC", nil, "SelfShockDuration") / 100
 				output.ShockEffectMod = skillModList:Sum("INC", cfg, "EnemyShockEffect")
 				if breakdown then
-					t_insert(breakdown.ShockDPS, s_format("For the minimum 5%% Shock to apply for 2 seconds, target must have no more than %d Ailment Threshold.", (((100 + output.ShockEffectMod)^(2.5)) * baseVal) / (100 * m_sqrt(10))))
+					t_insert(breakdown.ShockDPS, s_format("For the minimum 5%% Shock to apply for 2 seconds, target must have no more than %d Ailment Threshold.", ((((100 + output.ShockEffectMod)^(2.5)) * baseVal) / (100 * m_sqrt(10))) * output.ShockDurationMod))
```
Comment (NickRyder) [inline Modules/CalcOffence-3_0.lua]: Duration does not scale the ailment threshold. 

Duration does scale the line `Shock to apply for 2 seconds`

File: Modules/CalcOffence-3_0.lua [hunk 2]
```diff
@@ -2065,19 +2096,13 @@ function calcs.offence(env, actor, activeSkill)
 				sourceHitDmg = sourceHitDmg + output.LightningHitAverage
 				sourceCritDmg = sourceCritDmg + output.LightningCritAverage
 			end
-			local baseFreezeVal = calcAilmentDamage("Freeze", sourceHitDmg, sourceCritDmg)
-			local baseChillVal = calcAilmentDamage("Chill", sourceHitDmg, sourceCritDmg)
-			if baseFreezeVal > 0 then
+			local baseVal = calcAilmentDamage("Freeze", sourceHitDmg, sourceCritDmg)
+			if baseVal > 0 then
 				skillFlags.freeze = true
-				skillFlags.chill = true
```
Comment (NickRyder) [inline Modules/CalcOffence-3_0.lua]: This needs to stay in 
> When a target is frozen, it is also considered chilled.

File: Modules/CalcOffence-3_0.lua [hunk 3]
```diff
@@ -2049,11 +2046,45 @@ function calcs.offence(env, actor, activeSkill)
 				output.ShockDurationMod = 1 + skillModList:Sum("INC", cfg, "EnemyShockDuration") / 100 + enemyDB:Sum("INC", nil, "SelfShockDuration") / 100
 				output.ShockEffectMod = skillModList:Sum("INC", cfg, "EnemyShockEffect")
 				if breakdown then
-					t_insert(breakdown.ShockDPS, s_format("For the minimum 5%% Shock to apply for 2 seconds, target must have no more than %d Ailment Threshold.", (((100 + output.ShockEffectMod)^(2.5)) * baseVal) / (100 * m_sqrt(10))))
+					t_insert(breakdown.ShockDPS, s_format("For the minimum 5%% Shock to apply for 2 seconds, target must have no more than %d Ailment Threshold.", ((((100 + output.ShockEffectMod)^(2.5)) * baseVal) / (100 * m_sqrt(10))) * output.ShockDurationMod))
 					t_insert(breakdown.ShockDPS, s_format("^8(Ailment Threshold is about equal to Life except on bosses where is is about half of their life)"))
 				end
  			end
 		end
+		if output.ChillChanceOnHit > 0 then
```
Comment (NickRyder) [inline Modules/CalcOffence-3_0.lua]: You need to add in the `ChillChanceOnCrit` here.

File: Modules/CalcOffence-3_0.lua [hunk 4]
```diff
@@ -2049,11 +2046,45 @@ function calcs.offence(env, actor, activeSkill)
...
+				sourceHitDmg = sourceHitDmg + output.ChaosHitAverage
+				sourceCritDmg = sourceCritDmg + output.ChaosCritAverage
+			end
+			local baseVal = calcAilmentDamage("Chill", sourceHitDmg, sourceCritDmg)
+			if baseVal > 0 then
+				skillFlags.chill = true
+				output.ChillEffectMod = skillModList:Sum("INC", cfg, "EnemyChillEffect")
+				output.ChillDurationMod = 1 + skillModList:Sum("INC", cfg, "EnemyChillDuration") / 100
+				if breakdown then
+					t_insert(breakdown.ChillDPS, s_format("For the minimum 5%% Chill to apply for 2 seconds, target must have no more than %d Ailment Threshold.", ((((100 + output.ChillEffectMod)^(2.5)) * baseVal) / (100 * m_sqrt(10))) * output.ChillDurationMod))
```
Comment (NickRyder) [inline Modules/CalcOffence-3_0.lua]: This is wrong, duration doesnt scale this, see shock comment.

**Comment by Quotae** [comment]
Everything should be fixed now, I think.

**Comment by Quotae** [comment]
Just kidding, I had a typo I didn't notice or come across in my testing (which was pretty limited after I finished it yesterday besides checking that the duration display and equations were correct).

I also added support for Unbound Ailments' mod that gives increased effect of non-damaging ailments. This isn't in normal PoB either and is technically not the focus of this pull request, but it's close enough to related that I'm putting it here instead of making a new branch and PR. It gives increased effect of freeze for posterity, because the item/passive mod for increased effect of non-damaging ailments does.

**Pull request merged**